Project-Descriptions-and-Plans/LLVM-Toolchain/llvm-toolchain-pl.md

```diff
@@ -0,0 +1,314 @@
+- The PA describes a "subproject" for a particular target. Each PA should contain the full feature list and project plan for a particular target
+- There would normally be multiple PA sub-projects for top-level SW projects
+
+The flow for **OpenHW Advanced RISC-V Verification Methodology (ARVM)** subprojects:
+- The ARVM PC is written as a top-level project addressing multiple ARVM subprojects
+- The PL and PA describe an ARVM "subproject". The PA  contains the full feature list and project plan for the subproject
+- There will be multiple PL/PA subprojects for the ARVM top-level project
+
+**Part 2, PL fields:**
+The PL proposal explains the "what". Some of it can be updated directly from the PC proposal -->
+
+# Title of Project - "Clang/LLVM tool chain for CORE-V"
+# Project Launch Proposal
+## Date of proposal - 2022-11-28
+## Author(s) - Jeremy Bennett, Charlie Keaney
+
+## Summary of project
+
+This proposal is for a Clang/LLVM compiler tool chain for CORE-V.  Under the new program structure, this is the overall framework for all Clang/LLVM tool chain development for CORE-V.  Separate "Plan Approved" proposals will address individual target development.  As such there is minimal work directly associated with this Project Launch proposal.
+
+Support for the Software TG's primary goal of developing a thriving commercial ecosystem is a matter for the detailed planning of individual "Plan Approved proposals".
+
+In order to support the Software TG secondary goal of upstreaming all open source tool developments:
+
+- all Clang/LLVM development will be kept compliant with the LLVM Foundation coding and quality assurance standards;
+- the implementation will follow the upstream tool design and coding conventions; and
+- any vendor-specific modifications or additions will be duly isolated into extensions, so as to maintain a fully functional common open source code base.
+
+Finished LLVM-related work will be contributed to the LLVM community and maintained upstream.
+
+This project requires the modification of a set of existing public open source code bases maintained as projects of the LLVM Foundation.  Therefore, the processes used within OpenHW will reflect the processes of those upstream projects.
+
+The code base of LLVM is of substantial size:
+
+- llvm-project: 6.4 MLOC C++ including Clang, LLVM, LLDB, LLD.
+
+The LLVM code base includes substantial regression test suites (1.5 MLOC including `libc++` C++ library tests), and success with these test suites is a pre-requisite of upstream acceptance of any patch.
+
+LLVM only has unit tests up to IR level, the LLVM Integrated Tester (_lit_). There is a LLVM execution test suite, but it is of applications to run under an operating system, so only of use to tool chains targeting OS applications. For bare metal applications, the GCC regression tests are used. Execution tests in turn require availability of implementation targets against which the code can be run.
```

Comment:
"LLVM only has unit tests up to LLVM Intermediate Representation (IR) level, the LLVM Integrated Tester (_lit_)." 

First part of sentence implies there are multiple unit tests. Are you saying the _lit_ comprises multiple unit tests?  

It would be easier to understand if you said "comprised in the LLVM Integrated Tester (_lit_)"

Reply:
_lit_ is around 40,000 unit tests.